src/main/java/org/takes/facets/auth/PsBasic.java

```
@@ -0,0 +1,201 @@
+import org.takes.facets.flash.RsFlash;
+import org.takes.facets.forward.RsForward;
+import org.takes.misc.Opt;
+import org.takes.rq.RqHeaders;
+import org.takes.rs.RsWithHeaders;
+
+/**
+ * Pass that checks the user according RFC-2617.
+ *
+ * <p>The class is immutable and thread-safe.
+ *
+ * @author Endrigo Antonini (teamed@endrigo.com.br)
+ * @version $Id$
+ * @since 0.20
+ * @checkstyle ClassDataAbstractionCouplingCheck (250 lines)
+ */
+@EqualsAndHashCode(of = { "entry", "realm" })
+public final class PsBasic implements Pass {
+
+    /**
+     * Authorization response HTTP head.
+     */
+    private static final String AUTH_HEAD = "Basic";
+
+    /**
+     * Entry to validate user information.
+     */
+    private final transient PsBasic.Entry entry;
+
+    /**
+     * Realm.
+     */
+    private final transient String realm;
+
+    /**
+     * Ctor.
+     * @param rlm Realm
+     * @param basic Entry
+     */
+    public PsBasic(final String rlm, final PsBasic.Entry basic) {
+        this.realm = rlm;
+        this.entry = basic;
+    }
+
+    @Override
+    public Opt<Identity> enter(final Request request) throws IOException {
+        Opt<Identity> identity = new Opt.Empty<Identity>();
+        try {
+            final String decoded = new String(
+                DatatypeConverter.parseBase64Binary(
+                    new RqHeaders.Smart(
+                        new RqHeaders.Base(request)
+                    ).single("authorization").split(AUTH_HEAD)[1]
+                )
+            ).trim();
+            final String user = decoded.split(":")[0];
+            if (this.entry.enter(
+                user,
+                decoded.substring(user.length() + 1)
+            ).has()) {
+                identity = new Opt.Single<Identity>(
+                    new Identity.Simple(
+                        String.format("urn:basic:%s", user)
+                    )
+                );
+            }
+        } catch (final IOException ex) {
```

@antonini what is this for? why can't you just let the exception bubble up?